lib/api/pokedex_api_spec/handler/pokemon_handler.dart

```diff
@@ -18,7 +19,11 @@ class PokemonApi {
 
     final baseUri = Uri.parse(apiClient.dio.options.baseUrl);
     final uri = baseUri.replace(queryParameters: queryParams, path: '${baseUri.path}/pokemon');
-    return await apiClient.dio.getUri(uri).then(
-        (response) => response.data[str.results].map<Pokemon>((dynamic e) => Pokemon.fromJson(e as Json)).toList());
+    return await apiClient.dio.getUri(uri).then((response) => response.data[str.results]
+        .map<Pokemon>((dynamic pokemon) => Pokemon(
+            name: pokemon['name'],
+            url: pokemon['url'],
+            img: k.imageURL.replaceAll(str.indexReplace, pokemon['url'].toString().split('/')[6])))
```

Comment:
this isn't provided from the api response right? if you're gonna be needing this kind of string transformation I would recommend having a new dart file for extension functions 🙂